### PR TITLE
Extended export_xml() to allow reflective planes

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
@Shimwell what do you think?

Todo:
- docs
- add a check to ensure the difference between angles is below 180°

### CODE REVIEW

File: openmc_dagmc_wrapper/neutronics_model.py [hunk 1]
```diff
@@ -462,7 +464,51 @@ def export_xml(
...
+            reflective_1 = openmc.Plane(
+                a=sin(angle_1), b=-cos(angle_1), c=0.0, d=0.0,
+                surface_id=9995, boundary_type='reflective')
+            reflective_2 = openmc.Plane(
+                a=sin(angle_2), b=-cos(angle_2), c=0.0, d=0.0,
+                surface_id=9994, boundary_type='reflective')
+
+            region = -upper_x & -upper_y & +lower_z & - \
+                upper_z & -reflective_1 & +reflective_2
+            Q1_cell = openmc.Cell(cell_id=9999, region=region, fill=dag_univ)
```
Comment (Shimwell) [inline openmc_dagmc_wrapper/neutronics_model.py:509]:
```suggestion
            containing_cell = openmc.Cell(cell_id=9999, region=region, fill=dag_univ)
```

File: openmc_dagmc_wrapper/neutronics_model.py [hunk 2]
```diff
@@ -462,7 +464,51 @@ def export_xml(
...
+            root = [Q1_cell]
```
Comment (Shimwell) [inline openmc_dagmc_wrapper/neutronics_model.py:510]:
```suggestion
            root = [containing_cell ]
```

**Comment by Shimwell** [comment]
Looks like this will work for models that start at 0 degrees and are less that 180 degrees.

Do you think it is worth changing the vacuum surface to a sphere just to simplify the number of surfaces needed.

Then if we ever try to support models that don't start at 0 degrees this will be easier.

File: openmc_dagmc_wrapper/neutronics_model.py (hunk 2 above)
Comment (Shimwell) [reply openmc_dagmc_wrapper/neutronics_model.py:510]: a code inspector suggestion

**Comment by RemDelaporteMathurin** [comment]
This also supports models that don't start at zero degree. It's just that the difference between the two angles must be less than 180.

I think a sphere would solve problems such as "what is the third surface that closes the volume?" 👍

**Comment by Shimwell** [comment]
Hope you don't mind, I couldn't resits tinkering. It is still a bit messy but the logic should only make the vacuum surfaces if a graveyard is not found in the dagmc model

File: openmc_dagmc_wrapper/neutronics_model.py [hunk 3]
```diff
@@ -462,7 +498,65 @@ def export_xml(
         # this is the underlying geometry container that is filled with the
         # faceted DAGMC CAD model
         dag_univ = openmc.DAGMCUniverse(self.h5m_filename)
-        geom = openmc.Geometry(root=dag_univ)
+
+        if self.reflective_angles is None:
+            # if a graveyard is not found in the dagmc geometry a CSG one is
+            # made
+            if 'graveyard' not in di.get_materials_from_h5m(self.h5m_filename):
```
Comment (RemDelaporteMathurin) [inline openmc_dagmc_wrapper/neutronics_model.py:505]: @Shimwell it seems this line is failing in the tests

Comment (Shimwell) [reply openmc_dagmc_wrapper/neutronics_model.py:505]: ah yes, sorry I can see what I have messed up there. I shall fix now

**Comment by Shimwell** [comment]
tests are passing, I think we are just missing the > 180 degree check

**Comment by RemDelaporteMathurin** [comment]
> tests are passing, I think we are just missing the > 180 degree check

@Shimwell These have been added and the tests pass in circleCI now. Don't know about GA

**Comment by Shimwell** [comment]
Github actions appear to pass the tests and then crash so lets not worry about that

**Pull request merged**